components/org.wso2.carbon.dashboards.designer/src/main/fragments/dashboard-designer/public/js/common/widget-renderer.js

```diff
@@ -90,10 +93,19 @@ widget.renderer = {};
      * @type {{onSuccess: renderWidgetcallback.onSuccess, onFailure: renderWidgetcallback.onFailure}}
      */
     var renderWidgetcallback = {
+        widgetId: '',
         onSuccess: function (data) {
             var gadgetContainer = $(data)[0];
+            var id = "widget_" + gadgetContainer.getAttribute("data-grid-id");
```

Comment:
This can be simply access from the jQuery object `$(gadgetContainer).data('grid-id')`.